src/urlencoded.rs

```diff
@@ -1,11 +1,137 @@
-use std::collections::HashMap;
-use url::form_urlencoded;
 use groupable::Groupable;
+use hyper::uri::RequestUri;
+use hyper::uri::RequestUri::{Star, AbsoluteUri, AbsolutePath, Authority};
+use plugin::{Plugin, Pluggable};
+use request::Request;
+use status::StatusCode;
+use std::collections::HashMap;
+use std::io::{Error, Read};
+use typemap::Key;
+use url::{form_urlencoded, UrlParser};
+
+type QueryStore = HashMap<String, Vec<String>>;
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct Query(QueryStore);
```

Comment:
> then we should probably rename it.

Why? Query string may be used in URL and request body both. I don't see a reason to do this.
Finally, MultiMap is not clear IMHO. Maybe QueryMap? But is it worth?